homeassistant/components/directv/config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Config flow for DirecTV."""
+import logging
+import socket
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from DirectPy import DIRECTV
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_SERIAL
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DEFAULT_PORT
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+ERROR_CANNOT_CONNECT = "cannot_connect"
+ERROR_UNKNOWN = "unknown"
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+def get_ip(host: str) -> str:
+    """Translate hostname to IP address."""
+    if host is None:
+        return None  # pragma: no cover
+
+    return socket.gethostbyname(host)
+
+
+def get_dtv_version(host: str, port: int = DEFAULT_PORT) -> Any:
+    """Test the device connection by retrieving the receiver version info."""
+    # directpy does IO in constructor.
+    dtv = DIRECTV(host, port)
+    return dtv.get_version()
+
+
+async def validate_input(hass: HomeAssistantType, data: Dict) -> Dict:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    host = await hass.async_add_executor_job(get_ip, data["host"])
```

Comment:
Since we add two executor jobs in a row, it would make more sense to make `validate_input` a non-async function and run that in the executor.

Reply:
Also, why resolve the host to an IP ? That is less future proof?

Reply:
this was mostly a combination of the way I saw other integrations working and the parameters that directpy asks for. it seem though that hostnames work just as well so removing this logic!